tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/util/resolution/ResolutionData.java

```diff
@@ -30,4 +31,14 @@
     ExecutionEnvironmentResolutionHints getEEResolutionHints();
 
     Map<String, String> getAdditionalFilterProperties();
+
+    /**
+     * @return <code>true</code> if the resolve operation should fail if there are missing
+     *         requirements
+     */
+    boolean failOnMissingRequirements();
+
+    void addMissingRequirement(IRequirement requirement);
+
+    Collection<IRequirement> getMissingRequirements();
```

Comment:
This is not appropriate as part of the ResolutionData interface IMO. ResolutionData is to configure the resolution and shouldn't incorporate information about resolution results as the ResolutionData can be reused across multiple builds. Please rework it.

Reply:
> ResolutionData can be reused across multiple builds

There are only 3 locations where it is used, in all those locations the ResolutionData is disposed after the method call.